File: frontend/anguler-coupon-managment-system/src/app/services/admin.service.ts
import { Injectable } from '@angular/core';
import { NetService } from './net/net.service';
import  * as M from '../models/app.models';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class AdminService {

  constructor(private netService : NetService) { }

  addNewCompany(form: M.CompanyFormDetails ): Observable <any> {
    return this.netService.addNewCompany(form);
   }

   getAllCompanies() : Observable<M.CompanyFormDetails[]>{
     return this.netService.getAllCompanies();
   }

   getAllCustomers() : Observable<M.Customer[]>{
    return this.netService.getAllCustomers();
  }

   deleteCompany(companyId : number): Observable<any>{
     return this.netService.deleteCompany(companyId);
   }

   deleteCustomer(customerId : number): Observable<any>{
    return this.netService.deleteCustomer(customerId);
  }

  getCustomer(customerId : number) : Observable<M.Customer>{
    return this.netService.getCustomer(customerId);
  }

  updateCustomer(customerDetails : M.Customer) : Observable<any>{
    return this.netService.updateCustomer(customerDetails);
  }

  getCompany(companyId : number) : Observable<M.Company>{
    return this.netService.getCompany(companyId);
  }

  updateCompany(companyDetails : M.Company) : Observable<any>{
    return this.netService.updateCompany(companyDetails);
  }



}

File: backend/java-springBoot-coupon-system/src/main/java/com/coupon/Services/AdminService.java
package com.coupon.Services;

import org.springframework.http.ResponseEntity;

import com.coupon.entity.Company;
import com.coupon.entity.Customer;

import java.util.List;

public interface AdminService {

    public void addCompany (Company company) throws Exception;

    public void removeCompany (Company company) throws Exception;

    public void updateCompany (Company company) throws Exception;

    public Company getCompany(long companyId) throws Exception;

    public List<Company> getAllCompanies() throws Exception;

    public void addCustomer (Customer customer) throws  Exception;

    public void removeCustomer (Customer customer) throws Exception;

    public void updateCustomer (Customer customer) throws Exception;

    public Customer getCustomer(long customerId) throws Exception;

    public List<Customer> getAllCustomers() throws Exception;


}
